src/org/zaproxy/zap/extension/fieldenumeration/FieldEnumeration.java

```diff
@@ -0,0 +1,661 @@
+                        	}
+			if (success) {
+                                JOptionPane.showMessageDialog(View.getSingleton().getMainFrame(),
+                                                Constant.messages.getString("export.button.success"));
+                        }
+                	}
+
+			}
+		});
+		pack();
+		return result;
+	 }
+	
+	protected List<String> getColumnNames() {
+                List<String> columnNamesList = new ArrayList<>();
+                for (int col = 0; col < getTable().getColumnCount(); col++) {
+                        columnNamesList.add(getTable().getColumnModel().getColumn(col).getHeaderValue().toString());
+                }
+                return columnNamesList;
+        }
+
+	protected List<Object> getRowCells(int row) {
+                List<Object> cells = new ArrayList<>();
+                for (int col = 0; col < getTable().getColumnCount(); col++) {
+                        Object value = getTable().getValueAt(row, col);
+                        cells.add(value == null ? "" : value.toString());
+                }
+                return cells;
+        }
+
+	protected JTable getTable() {
+                return exportTable;
+        }
+
+        public void setTable(JTable table) {
+                this.exportTable = table;
+        }
+	
+	 private JPanel getJPanel() {
+		//this.setLayout(new BorderLayout());
+		Panel.setSize(new Dimension(600, 600));
+		Panel.setVisible(true);
+		tp.add("Field Enumeration", jPanel);
+		jPanel.setLayout(new GridBagLayout());
+		c1.fill = GridBagConstraints.HORIZONTAL;
+		jPanel.setSize(new Dimension(600, 600));
+		c1.gridx = 0;
+                c1.gridy = 0;
+		JLabel l1 = new JLabel("URL Form:");
+		l1.setFont(font);
+		jPanel.add(l1, c1);
+		c1.anchor = GridBagConstraints.NORTH;
+		c1.gridx = 1;
+                c1.gridy = 0;
+		jPanel.add(printURL, c1);
+		c1.gridwidth = 2;
+		c1.gridx = 0;
+                c1.gridy = 1;
+		JLabel l2 = new JLabel("Enter Regular Expression or status code");
+		l2.setFont(font);
+		jPanel.add(l2, c1);
+		c1.gridx = 0;
+                c1.gridy = 2;
+                //regex.setSize(100, 200);
+                jPanel.add(regex, c1);
+		c1.gridx = 0;
+                c1.gridy = 3;
+		JLabel l7 = new JLabel("Select set of Characters");
+                l7.setFont(font);
+                jPanel.add(l7, c1);
+		char_sets.addItem("US-ASCII");
+		char_sets.addItem("UTF-8");
+		char_sets.addItem("EBCIDIC");
+		c1.gridx = 0;
+                c1.gridy = 4;
+		jPanel.add(char_sets, c1);
+
+		Panel.add(tp);
+
+		final StringBuilder i_chars = new StringBuilder();		
+
+		buttonOK.addActionListener(new ActionListener() {
+		@Override
+
+			public void actionPerformed(ActionEvent event) {
+
+			field = (String) params.getSelectedItem();
+             		System.out.println("button/field selected: " + field);
+			
+                	selected_chars = (String) char_sets.getSelectedItem();
```

Comment:
Why not declare as a String instead of casting?

Reply:
Ditto.

Reply:
;)

Reply:
It is already defined as String